// This file is automatically generated by sila2lib.proto_builder.proto_builder version 0.2.0
// :generation date: 2020-10-08 11:17:41
//
// ---- PLEASE DO NOT MODIFY MANUALLY !! ---

syntax = "proto3";
import "SiLAFramework.proto";
package sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1;

// Feature: Control Loop Service
//   Allows to control a Qmix Device with a Control Loop
service ControlLoopService {
    // Write Set Point
    //   Write a Set Point value to the Controller Device
    rpc WriteSetPoint(sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.WriteSetPoint_Parameters) returns (sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.WriteSetPoint_Responses) {}
    // Run Control Loop
    //   Run the Control Loop
    rpc RunControlLoop(sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.RunControlLoop_Parameters) returns (sila2.org.silastandard.CommandConfirmation) {}
    rpc RunControlLoop_Info(sila2.org.silastandard.CommandExecutionUUID) returns (stream sila2.org.silastandard.ExecutionInfo) {}
    rpc RunControlLoop_Result(sila2.org.silastandard.CommandExecutionUUID) returns (sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.RunControlLoop_Responses) {}
    // Stop Control Loop
    //   Stops the Control Loop (has no effect, if no Loop is currently running)
    rpc StopControlLoop(sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.StopControlLoop_Parameters) returns (sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.StopControlLoop_Responses) {}
    // Controller Value
    //   The actual value from the Device
    rpc Subscribe_ControllerValue(sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.Subscribe_ControllerValue_Parameters) returns (stream sila2.de.cetoni.pumps.syringepumps.controlloopservice.v1.Subscribe_ControllerValue_Responses) {}
    
}

// ----------------- Data Type definitions -----------------


// ------ Command Parameter and Response definitions -------
message WriteSetPoint_Parameters {
    sila2.org.silastandard.Real SetPointValue = 1;
}

message WriteSetPoint_Responses {
    
}

message RunControlLoop_Parameters {
    
}

message RunControlLoop_Responses {
    
}

message StopControlLoop_Parameters {
    
}

message StopControlLoop_Responses {
    
}
message Subscribe_ControllerValue_Parameters {
    // Empty message
}

message Subscribe_ControllerValue_Responses {
    sila2.org.silastandard.Real ControllerValue = 1;
}

// ----------------- Metadata Definitions ------------------
